Added indications of active layers in foldbutton and individual menu items

HarryMytilinaios opened the pull request:
This branch adds indications in the left panel of active layers enabled to individual accordion dropdown items, to layers tab and to the fold button when the sidebar is closed. Also with the refactoring of some of the `SwitchItem` code I managed to fix the transparency slider bug when you toggle a layer reduce transparency and enable it again the transparency didn't change. Please let me know if this is what we wanted as an expected behavior. Cc @wadhwamatic @ericboucher . Below I attach some screenshots of the indications. The UX is the following: When a user is on the layers tab the specific information of the active layer is displayed on the specific item that is enabled. When a user switches the tab and goes for example to the charts panel or the analysis panel, a badge with the number of the active layers are displayed on the header tab of the layers. When the sidebar is folded and some of the layers are active a badge is displayed above the icon of the FoldButton indicating how many active layers are enabled.

Screenshot/video of feature:
![Screenshot from 2023-04-20 13-20-46](https://user-images.githubusercontent.com/62129256/233338642-d8bae0c5-9d48-421c-b3d7-168909ee6cad.png)
![Screenshot from 2023-04-20 13-20-53](https://user-images.githubusercontent.com/62129256/233338685-94fafac9-e9d4-4d7d-a1d2-b5d5dd870a33.png)
![Screenshot from 2023-04-20 13-21-03](https://user-images.githubusercontent.com/62129256/233338730-8a8afbd9-eeb7-417a-a648-d1a39c676790.png)

ericboucher commented:
Great work @HarryMytilinaios 
This touches on quite a few components, can you give Amit and I a few pointers on what to test for optimal QA?
Layers, tabs, ...  Thanks!

HarryMytilinaios commented:
Of course. In this branch we added indications about the active enabled layers in the sidebar. There are 3 different types of indications.
- When a user is on the layers tab (In the screenshots above). Every layer group and individual layer item has a `Chip` component that indicates how many layers are enabled on the specific layer group and the specific individual component. You toggle and untoggle mutliple layers and the indications will follow the number of active ones.
- When the user switches to a different tab (e.g `Charts`, `Analysis`), then a small `Badge` with the number of active layers will be displayed on the tab title at the left top of the icon, when the user switch back to the `Layers` tab the `Badge` will dissapear from the tab title and the individual information of the tabs will be displayed on the individual layer categories and layer category items.
- At last when the hide the sidebar at all then in the toggle button that shows and hides the sidebar will be displayed a badge with the summary number of all the active layers enabled.

Also in this branch is fixed the opacity slider problem, (When you toggled a layer, reduced opacity, untoggled it and toggle it again), would create inconsistency in the UI and the actual value of the opacity. This is now fixed by resetting the opacity every time the layer is untoggled.

wadhwamatic commented:
Great work @HarryMytilinaios! The functionality covers exactly what is needed. 

The text '1 Active Layer' on the layer group header is too prominent. I think the badge approach, just showing the number would work well here and on-hover, show the full text 'x Active Layer(s)'. 

I didn't test to see if the text introduced in this features is translatable. If it's not yet, please add that too and I can test it when ready.

HarryMytilinaios commented:
@wadhwamatic thank you for your review. I fixed the active layers indicators to only show the number of active layers and if you hover to show the full information `x Active Layer(s)`. The text is translatable we only need to add translation for the label `Active Layer(s)`. Also I now I saw the problem with the transparency and the different toggling of layers. I managed to fix that and now the opacity is always reset

ericboucher commented on frontend/src/context/mapStateSlice/selectors.ts:
```diff
@@ -5,11 +5,15 @@ import { Map as MapBoxMap } from 'mapbox-gl';
   state.mapState.layers;
+export const activeLayersSelector = (state: RootState): MapState['layers'] =>
+  state.mapState.layers.filter((layer: LayerType) => {
+    return !layer.id.includes('boundaries');
```
you can use `layer.type !== 'boundary'` instead which is safer

HarryMytilinaios replied:
Of course.

wadhwamatic commented:
> @wadhwamatic thank you for your review. I fixed the active layers indicators to only show the number of active layers and if you hover to show the full information `x Active Layer(s)`. The text is translatable we only need to add translation for the label `Active Layer(s)`. Also I now I saw the problem with the transparency and the different toggling of layers. I managed to fix that and now the opacity is always reset

@HarryMytilinaios - all looks good now - thanks!
